src/main/grammars/RustParser.bnf

```diff
@@ -1319,41 +1320,39 @@ Macro2 ::= AttrsAndVis MACRO_KW identifier ( Macro2FunctionLikeBody | Macro2Matc
 private Macro2FunctionLikeBody ::= '(' MacroPatternContents ')' '{' MacroExpansionContents '}'
 private Macro2MatchLikeBody ::= '{' (MacroCase ','?)* '}'
 
-MacroCall ::= MacroHead <<macroSemicolon>> {
+fake MacroCall ::= AttrsAndVis PathWithoutTypes '!' identifier? (
+    MacroArgument | ExprMacroArgument | FormatMacroArgument | AssertMacroArgument |
+      VecMacroArgument | LogMacroArgument | IncludeMacroArgument | ConcatMacroArgument
+    ) ';'? {
   implements = [ "org.rust.lang.core.psi.ext.RsOuterAttributeOwner"
                  "org.rust.lang.core.macros.RsExpandedElement"
                  "org.rust.lang.core.psi.ext.RsModificationTrackerOwner" ]
-  pin = 2
-  hooks = [ leftBinder = "ADJACENT_LINE_COMMENTS" ]
   mixin = "org.rust.lang.core.psi.ext.RsMacroCallImplMixin"
   stubClass = "org.rust.lang.core.stubs.RsMacroCallStub"
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
-MacroCallNoSemicolons ::= MacroHead {
-  pin = 2
+ItemLikeMacroCall ::= <<parseMacroCall 'MC_ITEM'>> {
   elementType = MacroCall
   hooks = [ leftBinder = "ADJACENT_LINE_COMMENTS" ]
+  name = "macro call"
 }
 
-private NonSpecialMacroCall ::= &(!SpecialExprContextMacroNames) MacroCall
-
-// We parse some macros (like `println`) as a specific syntax constructions. This
-// is a list of such specific macros that should be always parsed as `MacroExpr`.
-// E.g. `println!();` should be parsed as `ExprStmt(MacroExpr(MacroCall(...)))`, but
-// any other `foo!();` as a simple `MacroCall(...)`
-private SpecialExprContextMacroNames ::= "try" | "await" | "dbg" | "format" | "format_args" | "write" |
-"writeln" | "print" | "println" | "eprint" | "eprintln" | "panic" | "unimplemented" | "unreachable" |
-"assert" | "debug_assert" | "assert_eq" | "assert_ne" | "debug_assert_eq" | "debug_assert_ne" |
-"trace" | "log" | "warn" | "debug" | "error" | "info" | "include_str" | "include_bytes" | "concat"
+BlockElementMacroCall ::= <<parseMacroCall 'MC_BLOCK'>> {
+  elementType = MacroCall
+  hooks = [ leftBinder = "ADJACENT_LINE_COMMENTS" ]
+  name = "macro call"
+}
 
-private MacroHead ::= AttrsAndVis &(PathWithoutTypes '!') (SpecialMacro | CommonMacro)
+ExprLikeMacroCall ::= <<parseMacroCall 'MC_EXPR'>> {
```

Comment:
Hm, as we discussed in person, we need `MC` prefix only to avoid duplicate names in the generated parser.
I personally don't like this prefix because it makes me think about what it means. Of course, it means `macro call` but this knowledge is quite unusable because when I'm reading this rule I'm already in macro call context.
Maybe it's better just to replace it with 
```suggestion
ExprLikeMacroCall ::= <<parseMacroCall 'MacroCallParsingMode.EXPR'>> {
```
and drop prefix at all?